secrets/secrets.go

```diff
@@ -321,21 +323,47 @@ func secretsValidate(secretsDocument Document) (*Secrets, error) {
 	return secrets, nil
 }
 
+type notCSIError struct {
+	err error
+}
+
+func (e notCSIError) Error() string {
+	return fmt.Sprintf("configured directory does not appear to be a Vault CSI directory: %v", e.err)
+}
+
+func (e notCSIError) Unwrap() error {
+	return e.err
+}
+
 // walkCSIDirectory parses a directory for vault secrets and merges them into one object
 func walkCSIDirectory(dir fs.FS) (Document, error) {
+	const (
+		// This is where k8s actually writes the content,
+		// ref: https://pkg.go.dev/sigs.k8s.io/secrets-store-csi-driver/pkg/util/fileutil#AtomicWriter.Write
+		k8sSubdirectory = "..data"
+	)
 	secretsDocument := Document{
 		Secrets: make(map[string]GenericSecret),
 	}
 	err := fs.WalkDir(
 		dir,
-		".",
+		k8sSubdirectory,
 		func(path string, d fs.DirEntry, err error) error {
 			if err != nil {
+				if path == k8sSubdirectory && errors.Is(err, fs.ErrNotExist) {
+					return notCSIError{err: err}
+				}
 				return err
 			}
 			if d.IsDir() {
 				return nil
 			}
+			relPath, err := filepath.Rel(k8sSubdirectory, path)
+			if err != nil {
+				// Should not happen as this means path is not under k8sSubdirectory,
+				// but just in case
+				return nil
```

Comment:
If we handle it for _just in case_, we should also just return the actual error, shouldn't we?

Reply:
that would actually stop the walking and cause the whole parser to fail, which has bigger consequences, so probably not. but I don't feel too strongly either way